### ISSUE TITLE
Building succeeds but loading library throws error

### ISSUE THREAD

**Issue by gwd999** [klausspanderen/RHestonSLV#2]
The build now against R-4.0.5 on Ubuntu 20.04 (after commit 8a169a3878c37b80fb3ecbe45f990db82b89b1f9) runs smoothly 
But after starting R-4.0.5 and loading library below behavior can be observed: 
```
❯ library(RHestonSLV)
Error: package or namespace load failed for ‘RHestonSLV’:
 .onLoad failed in loadNamespace() for 'RHestonSLV', details:
  call: getQuantLibVersion()
  error: could not find function "getQuantLibVersion"
```
I have also noticed that in files `Makevars` and `Makvars.in` 
```
PKG_LIBS=$(shell "Rscript" -e "Rcpp:::LdFlags()") -lQuantLib -lrt 
```
is used in which `Rcpp:::LdFlags()` is a deprecated Rcpp function (as of Rel. 0.12.19) and returns an empty string. It seems to me that somehow the build package is not picking up the Rcpp exported `getQuantLibVersion()` and my guess is that the PKG_LIBS do not get linked correctly?  

By commenting out the whole `getQuantLibVersion()` function part in `zzz.R` the library manages to load, but currently I have not succeeded to produce results with any of the demo code. 
 
 And one final minor detail (more like an inconsistency nothing really serious) I noted in `zzz.R`
 ```
qlver <- getQuantLibVersion()
  if (compareVersion(qlver, "1.12") < 0) {
    packageStartupMessage("QuantLib version ", qlver, " detected which is older than 1.8.")
... 
```
there is a comparison against `"1.12"` and the message following it "still" mentions `version 1.8`

**Comment by gwd999** [comment]
Some results from running the demos ... 

```
demo(topic = 'impliedvol', package = 'RHestonSLV')
... 
❯ modelmc <- new (HestonSLVMCModel,
+                 Sys.Date(), Sys.Date()+365,
+                 localVol, process, mcParams)
Error in new_CppObject_xp(fields$.module, fields$.pointer, ...) :
  std::bad_alloc
```
The above being the most common error. 
Second type of error appearing in barrierprice demo
```
demo(topic = 'barrierprice', package = 'RHestonSLV')
... 
❯ prices <- sapply(c(1.0,0.5,0.25, 0.1), function(eta) {
+   slvBarrierPrices(eta)
+ })
Error in model$leverageFunction(t, strike) : std::bad_alloc
```
and the `moustache` demo just hangs itself on line  89: 
```
❯ implVol = hestonSLVOptionPricer(Sys.Date(), 100, "call", "european", Sys.Date()+365, process, function(s, t) { 1.0 }, 101, 401, 101, 0, "ModifiedCraigSneyd")$impliedVol
```

**Comment by klausspanderen** [comment]
Hi

I've just installed the latest R version 4.1.1 on Ubuntu 20.04 LTS (https://rtask.thinkr.fr/installation-of-r-4-0-on-ubuntu-20-04-lts-and-tips-for-spatial-packages/) and the build - including the demos - is working. 

W.r.t. the problems with getQuantLibVersion, the RQuantLib library has to be included with library(RQuantLib) before you add library(RHestonSLV). At least for me this did the trick. 

I've parallelized all demos because they can take quite some time to run (e.g. 10min). Thx for the hint with the version number!

BTW, all the SLV functionality is included in Python via the standard QuantLib package. If you can switch to Python then you can skip the compilation issues with R.

**Comment by gwd999** [comment]
> BTW, all the SLV functionality is included in Python via the standard QuantLib package. If you can switch to Python then you can skip the compilation issues with R.

Thanks for the hint - since reticulate is available for some time now - this is definitely a valid option. 

Thumbs up 👆 all clear from my side as well - potentially rebuilding QuantLib-1.17 last night AND setting `./configure --enable-intraday` , since afair RQuantLib has that set by default since forever - could have done the trick @klausspanderen: maybe that configuration is a valid candidate for your README? 
Plus I also made sure this time to load RQuantLib first in line as you suggested - maybe also missed that last night (was quite late already here in AUT). 

I did underestimate the necessary demo times (thnx for  the hint) by a huge factor ... 
The 'barrierdelta' demo took me approx. 10-12 minutes to deliver the beauty below on a beefed up XPS13 7390 (32GB of RAM, 4 cores i7-1065G7@1.3GHz)  - so quite time consuming even on a "fairly decent" machine plus I missed out on activating the parallelization yesterday.

![image](https://user-images.githubusercontent.com/306826/135500328-12c5ee45-3c63-412e-b067-83c899c02024.png)
 
@klausspanderen - again a final big 'Thank you' for the update/assistance and this cool package!
